src/OpenApi/test/Services/OpenApiDocumentServiceTests.Paths.cs

```diff
@@ -0,0 +1,158 @@
+                        });
+                });
+        });
+    }
+
+    [Fact]
+    public async Task GetOpenApiPaths_RespectsShouldInclude()
+    {
+        // Arrange
+        var builder = CreateBuilder();
+
+        // Act
+        builder.MapGet("/api/todos", () => { }).WithMetadata(new EndpointGroupNameAttribute("v1"));
+        builder.MapGet("/api/users", () => { }).WithMetadata(new EndpointGroupNameAttribute("v2"));
+
+        await VerifyOpenApiDocument(builder, document =>
+        {
+            Assert.Collection(document.Paths.OrderBy(p => p.Key),
+                path =>
+                {
+                    Assert.Equal("/api/todos", path.Key);
+                }
+            );
+        });
+    }
+
+    [Fact]
+    public async Task GetOpenApiPaths_RespectsSamePaths()
+    {
+        // Arrange
+        var builder = CreateBuilder();
+
+        // Act
+        builder.MapGet("/api/todos", () => { });
+        builder.MapPost("/api/todos", () => { });
+
+        // Assert
+        await VerifyOpenApiDocument(builder, document =>
+        {
+            Assert.Collection(document.Paths.OrderBy(p => p.Key),
+                path =>
+                {
+                    Assert.Equal("/api/todos", path.Key);
+                    Assert.Collection(path.Value.Operations.OrderBy(o => o.Key),
+                        operation =>
+                        {
+                            Assert.Equal(OperationType.Get, operation.Key);
+                        },
+                        operation =>
+                        {
+                            Assert.Equal(OperationType.Post, operation.Key);
+                        });
+                }
+            );
+        });
+    }
+
+    [Fact]
+    public async Task GetOpenApiPaths_HandlesRouteParameters()
+    {
+        // Arrange
+        var builder = CreateBuilder();
+
+        // Act
+        builder.MapGet("/api/todos/{id}", () => { });
+        builder.MapPost("/api/todos/{id}", () => { });
+
+        // Assert
+        await VerifyOpenApiDocument(builder, document =>
+        {
+            Assert.Collection(document.Paths.OrderBy(p => p.Key),
+                path =>
+                {
+                    Assert.Equal("/api/todos/{id}", path.Key);
+                    Assert.Collection(path.Value.Operations.OrderBy(o => o.Key),
+                        operation =>
+                        {
+                            Assert.Equal(OperationType.Get, operation.Key);
+                        },
+                        operation =>
+                        {
+                            Assert.Equal(OperationType.Post, operation.Key);
+                        });
+                }
+            );
+        });
+    }
+
+    [Fact]
+    public async Task GetOpenApiPaths_HandlesRouteConstraints()
+    {
+        // Arrange
+        var builder = CreateBuilder();
+
+        // Act
+        builder.MapGet("/api/todos/{id:int}", () => { });
```

Comment:
Does having two different verbs for the route improve code coverage?  It seems like it just combines different-verbs (covered above) with route constraints (actual test).

Reply:
It doesn't impact code coverage in a major way. I did it because in real-world scenarios having two endpoints with the same HTTP method and route will produce a runtime exception so this test more closely matches how shared paths will show up in user code.